File: Domain/Motofy.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Motofy
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public virtual AppUser Publisher { get; set; }
        public virtual Brand Brand { get; set; }
        public string Model { get; set; }
        public string CubicCentimeters { get; set; }
        public string PhotoUrl { get; set; }
        public string Description { get; set; }
        public string YearOfProduction { get; set; }
        public DateTime DatePublished { get; set; }
        public string City { get; set; }
        public virtual Country Country { get; set; }
        public string PricePaid { get; set; }
        public string EstimatedValue { get; set; }
        public string NumberOfKilometers { get; set; }
        public int TotalEmbraced { get; set; }
        public virtual AverageRating AverageRating { get; set; }
        public virtual ICollection<UserMotofy> UserMotofies { get; set; }
        public virtual MotofyPhoto MotofyPhoto { get; set; }
        public virtual ICollection<CommentMotofy> CommentMotofies { get; set; }
        public virtual ICollection<MotofyScore> MotofyScores { get; set; }
    }
}

File: client-app/src/app/models/motofy.ts
import { IComment } from "./comment";

export interface IMotofyEnvelope {
    motofies: IMotofy[];
    motofyCount: number;
    mostEmbraced: IMotofy;
    highestRatedMotofy: IMotofy;
}
export interface IMotofy {
    id: string;
    name: string;
    brandId: string;
    brandName: string;
    brandLogoUrl: string;
    model: string;
    cubicCentimeters: string;
    photoUrl?: string | null;
    description?: string;
    yearOfProduction?: string;// Date;
    datePublished?: string;
    city: string;

    countryName: string;
    countryId: string;

    pricePaid: string;
    estimatedValue: string;
    numberOfKilometers: string; 
    embraced: boolean;
    isOwner: boolean;
    embracers: IEmbracer[];
    commentMotofies: IComment[];
    file: Blob;

    motofyScores: any[];
    averageRating: string;
}

export interface IMotofyFormValues extends Partial<IMotofy>{//, Partial<IBrand> 

}

export class MotofyFormValues implements IMotofyFormValues {
    id?: string = '';
    name: string = '';
    // brand: string= '';
    brandId: string= '';
    brandName: string= '';
    model: string= '';
    cubicCentimeters: string= '';
    photoUrl: string  = '';
    description: string= '';
    yearOfProduction: string= '';
    // datePublished?: Date = undefined;
    datePublished?: string= '';
    city: string= '';
    countryName: string= '';
    countryId: string= '';
    pricePaid: string= '';
    estimatedValue: string= '';
    numberOfKilometers: string= '';
    // file: IPhoto;
    // brands?: IBrand[];

    constructor(init?: MotofyFormValues ) {//, brandsToSelect?: IBrand[]
        if (init ) {//&& brandsToSelect
            // init.brands = brandsToSelect;
        }
        Object.assign(this, init);
    }
} 

export interface IEmbracer {
    username: string;
    displayName: string;
    image: string;
    isOwner: boolean;
    following?: boolean;
}

export interface IRateMotofy {
    id: string;
    score: string | number | undefined;
}

export interface IMotofyScore {
    username: string | undefined;
    displayName: string | undefined;
    score: string | number | undefined;
}

